src/bitbox02.js

```diff
@@ -15,16 +15,28 @@ function sleep(ms) {
 export async function getDevicePath() {
     const attempts = 10;
     for (let i=0; i<attempts; i++){
-        let response;
+        let deviceResponse;
+        let errorMessage;
         try {
-            response = await fetch("http://localhost:8178/api/v1/devices", {
+            await fetch("http://localhost:8178/api/v1/devices", {
                 method: 'GET',
                 headers: {},
+            })
+            .then(response => {
```

Comment:
You are kind of mixing `async/await` and futures here. Is it possible to have `catch(err)` and then look at `err` for the status code instead?

Reply:
You're right, the `.then` is not needed with the `await`, but you can't catch it with `catch`, because it won't actually be an error if you get a response like 403, so you do have to look at the `response.ok` and `response.status` in the `try` block, not in `catch`, that won't be reached. 

Changed it to just be handled without the `.then`